### ISSUE TITLE
CNL2.4 driver does not work reliably for pump status downloads

### ISSUE THREAD

**Issue by ondrej1024** [ondrej1024/ddguard#1]
While the CNL2.4 driver is running quite reliably on an RPi3 (no hangups for a day or so), on the RPi-Zero I get only 1 good reading from the CNL. At the second attempt I get this error:

```
2019-11-23 19:20:52,206 WARNING [read_minimed_next24] Timeout waiting for message
2019-11-23 19:20:52,253 ERROR [read_minimed_next24]  ### checkControlMessage: Expected to get an 0x5 control character, got message with length 293 and control char 0x125

```
At the next reading I get:

```
2019-11-23 19:20:57,306 ERROR [read_minimed_next24] readDeviceInfo: Expected to get an ASTM message, but got 05 instead
```
and the CNL hangs up compeltely, which means it even disappears from the USB bus.

**Comment by ondrej1024** [comment]
Work to fix this is ongoing in the [comms_robustness_backport](https://github.com/ondrej1024/ddguard/tree/comms_robustness_backport) branch

**Comment by ondrej1024** [comment]
Some of the communication robustness code from the [Android uploader](https://github.com/pazaan/600SeriesAndroidUploader) has been backported to the Python driver lib, as suggested by @Pogman in [this discussion](https://github.com/pazaan/600SeriesAndroidUploader/issues/261#issuecomment-557530520)

Tests with these improvements show that the DD-Guard gateway now runs for a whole day without any CNL error. No USB port power cycle is necessary to recover the CNL stick. Therefore the gateway can run reliably also on a RPi-Zero where USB power cannot be controlled.

This has been released as [V0.5](https://github.com/ondrej1024/ddguard/releases/tag/V0.5)
